src/openfermion/ops/_polynomial_tensor.py

```diff
@@ -221,11 +223,17 @@ def __add__(self, addend):
         summand += addend
         return summand
 
-    def __neg__(self):
-        neg_n_body_tensors = dict()
+    def with_function_applied_elementwise(self, func):
+        new_n_body_tensors = dict()
         for key in self.n_body_tensors:
-            neg_n_body_tensors[key] = numpy.negative(self.n_body_tensors[key])
-        return PolynomialTensor(neg_n_body_tensors)
+            new_n_body_tensors[key] = func(self.n_body_tensors[key])
+        return PolynomialTensor(new_n_body_tensors)
+
+    def __neg__(self):
+        return self.with_function_applied_elementwise(operator.neg)
+
+    def __mod__(self, other):
+        return self.with_function_applied_elementwise(lambda x: x % other)
```

Comment:
Do you have an actual use case for modding a PolynomialTensor? Seems quite unnatural to me ...

Reply:
Yes, I have some code that exponentiates an InteractionOperator that I test by comparing it the the log of the exponentiated unitary.